src/main/scala/cromwell/binding/expression/FileEvaluator.scala

```diff
@@ -0,0 +1,93 @@
+
+  private def evalValue(ast: AstNode): Try[WdlValue] = valueEvaluator.evaluate(ast)
+
+  private def evalValueToWdlFile(ast: AstNode): Try[Seq[WdlFile]] = {
+    evalValue(ast) match {
+      case Success(p:WdlPrimitive) => Success(Seq(WdlFile(p.valueString)))
+      case Success(_) => Failure(new WdlExpressionException(s"Expecting a primitive type from AST:\n${ast.toPrettyString}"))
+      case Failure(e) => Failure(e)
+    }
+  }
+
+  private def findWdlFiles(value: WdlValue, coerce: Boolean = true): Seq[WdlFile] = {
+    val coercedValue = if (coerce) coerceTo.coerceRawValue(value) else Success(value)
+    coercedValue match {
+      case Success(f: WdlFile) => Seq(f)
+      case Success(a: WdlArray) =>
+        a.value.flatMap(findWdlFiles(_, coerce=false))
+      case Success(m: WdlMap) =>
+        m.value flatMap { case (k, v) => Seq(k, v) } flatMap(findWdlFiles(_, coerce=false)) toSeq
+      case _ => Seq.empty[WdlFile]
+    }
+  }
+
+  override def evaluate(ast: AstNode): Try[Seq[WdlFile]] = {
+    /**
+     * First check if the top-level expression evaluates to a
+     * literal value.  If it does, return all WdlFiles referenced
+     * in that literal value
+     */
+    valueEvaluator.evaluate(ast) match {
+      case Success(v) => Success(findWdlFiles(v))
+      case Failure(ex) => _evaluate(ast)
+    }
+  }
+
+  private def _evaluate(ast: AstNode): Try[Seq[WdlFile]] = {
+    ast match {
+      case a: Ast if a.isFunctionCallWithOneFileParameter =>
+        evalValueToWdlFile(a.params.head) match {
+          case Success(v) => Success(v)
+          case _ => _evaluate(a.params.head)
+        }
+      case a: Ast if a.isBinaryOperator =>
+        evalValueToWdlFile(a) match {
+          case Success(f:WdlFile) => Success(Seq(f))
+          case _ => (_evaluate(a.getAttribute("lhs")), _evaluate(a.getAttribute("rhs"))) match {
+            case (Success(a:Seq[WdlFile]), Success(b:Seq[WdlFile])) => Success(a ++ b)
+            case _ => Failure(new WdlExpressionException(s"Could not evaluate:\n${a.toPrettyString}"))
+          }
+        }
+      case a: Ast if a.isUnaryOperator =>
+        evalValueToWdlFile(a) match {
+          case Success(f:WdlFile) => Success(Seq(f))
+          case _ =>
+            _evaluate(a.getAttribute("expression")) match {
+              case Success(a:Seq[WdlFile]) => Success(a)
+              case _ => Failure(new WdlExpressionException(s"Could not evaluate:\n${a.toPrettyString}"))
+            }
+        }
+      case a: Ast if a.isArrayOrMapLookup =>
+        evalValue(a) match {
+          case Success(f:WdlFile) => Success(Seq(f))
+          case _ => _evaluate(a.getAttribute("rhs"))
+        }
+      case a: Ast if a.isMemberAccess =>
+        evalValue(a) match {
+          case Success(f:WdlFile) => Success(Seq(f))
+          case _ => Success(Seq.empty[WdlFile])
+        }
+      case a: Ast if a.isArrayLiteral => ???
+      case a: Ast if a.isMapLiteral => ???
```

Comment:
They could be, though not implementing them will not take away functionality.  I can implement it now if you want